lib/datadog/core/environment/variable_helpers.rb

```diff
@@ -24,10 +24,34 @@ def env_to_float(var, default = nil)
           var && ENV.key?(var) ? ENV[var].to_f : default
         end
 
-        def env_to_list(var, default = [])
+        # Parses comma- or space-separated lists.
+        #
+        # If a comma is present, then the list is considered comma-separated.
+        # Otherwise, it is considered space-separated.
+        #
+        # After the entries are separated, commas and whitespaces that are
+        # either trailing or leading are trimmed.
+        #
+        # Empty entries, after trimmed, are also removed from the result.
+        def env_to_list(var, default = [], comma_separated_only: false)
```

Comment:
Minor: Would it perhaps make sense to make this a required argument, to make callers explicitly think about if they need commas or spaces for separation?

```suggestion
        def env_to_list(var, default = [], comma_separated_only:)
```

Reply:
Sure.